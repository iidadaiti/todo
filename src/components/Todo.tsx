import React, { ComponentProps, FC } from 'react'
import { Table, TableBody } from './Table'
import { TodoItem, TodoItemProps } from './TodoItem'

export interface TodoData {
  key: string
  text: string
  isDone: boolean
  createAt: Date
}

export interface TodoProps
  extends Omit<ComponentProps<'table'>, 'children'>,
    Pick<TodoItemProps, 'onRemoveTodo' | 'onChangeTodo'> {
  todoList: TodoData[]
}

export const Todo: FC<TodoProps> = ({
  todoList,
  onRemoveTodo,
  onChangeTodo,
  ...props
}) => (
  <Table {...props}>
    <TableBody>
      {todoList.map((data) => (
        <TodoItem
          key={data.key}
          data={data}
          onChangeTodo={onChangeTodo}
          onRemoveTodo={onRemoveTodo}
        ></TodoItem>
      ))}
    </TableBody>
  </Table>
)
