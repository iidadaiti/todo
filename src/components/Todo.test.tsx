import { render } from '@testing-library/react'
import React from 'react'
import { Todo, TodoProps } from './Todo'

test('Todo snapshot', () => {
  const todoList: TodoProps['todoList'] = [
    { key: '01', text: '', isDone: false, createAt: new Date() }
  ]

  const result = render(<Todo todoList={todoList}></Todo>)
  expect(result.asFragment()).toMatchSnapshot()
})
