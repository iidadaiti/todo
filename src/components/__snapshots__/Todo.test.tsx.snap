// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo snapshot 1`] = `
<DocumentFragment>
  <table
    class="spectrum-Table spectrum-Table--sizeM"
  >
    <tbody
      class="spectrum-Table-body"
    >
      <tr
        class="spectrum-Table-row"
      >
        <td
          class="spectrum-Table-cell"
          valign="middle"
        >
          <input
            id="todo-item-01"
            type="checkbox"
          />
        </td>
        <td
          class="spectrum-Table-cell textCell"
          valign="middle"
        />
        <td
          class="spectrum-Table-cell"
          valign="middle"
        >
          <button
            class="spectrum-Button spectrum-Button--cta spectrum-Button--sizeM"
            disabled=""
          >
            <span
              class="spectrum-Button-label"
            >
              Remove
            </span>
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
